From 7f6fa69221c41a3cf81c4f1bcba45792baaba7d7 Mon Sep 17 00:00:00 2001
From: Willy Tarreau <w@1wt.eu>
Date: Tue, 23 Apr 2013 19:39:43 +0200
Subject: [PATCH 29/37] BUG/MINOR: fix unterminated ACL array in compression

Recent commit 727db8b4 was lacking a NULL ACL descriptor to terminate
the array, causing a random behaviour upon startup. No backport is needed.
---
 src/compression.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/compression.c b/src/compression.c
index e75f21e..906e4fd 100644
--- a/src/compression.c
+++ b/src/compression.c
@@ -637,6 +637,7 @@ smp_fetch_res_comp_algo(struct proxy *px, struct session *l4, void *l7, unsigned
 /* Note: must not be declared <const> as its list will be overwritten */
 static struct acl_kw_list acl_kws = {{ },{
 	{ "res.comp",          NULL,            acl_parse_nothing,     acl_match_nothing  },
+	{ /* END */ },
 }};
 
 /* Note: must not be declared <const> as its list will be overwritten */
-- 
1.8.1.5

